kyo-core/shared/src/test/scala/kyoTest/ResultTest.scala

```diff
@@ -0,0 +1,301 @@
+package kyoTest
+
+import kyo.*
+import kyo.Result.Failure
+import kyo.Result.Success
+
+class ResultTest extends KyoTest:
+
+    val try2: Result[Result[Int]] = Success(Failure(new Exception("error")))
+
+    "should match Success containing Failure" in {
+        val result = try2 match
+            case Success(Failure(e)) => e.getMessage
+            case _                   => ""
+        assert(result == "error")
+    }
+
+    "isSuccess" - {
+        "returns true for Success" in {
+            assert(Success(1).isSuccess)
+        }
+        "returns false for Failure" in {
+            assert(!Failure[Int](new Exception("error")).isSuccess)
+        }
+    }
+
+    "isFailure" - {
+        "returns false for Success" in {
+            assert(!Success(1).isFailure)
+        }
+        "returns true for Failure" in {
+            assert(Failure[Int](new Exception("error")).isFailure)
+        }
+    }
+
+    "get" - {
+        "returns the value for Success" in {
+            assert(Success(1).get == 1)
+        }
+        "throws the exception for Failure" in {
+            val ex = new Exception("error")
+            assertThrows[Exception](Failure[Int](ex).get)
+        }
+    }
+
+    "getOrElse" - {
+        "returns the value for Success" in {
+            assert(Success(1).getOrElse(0) == 1)
+        }
+        "returns the default value for Failure" in {
+            assert(Failure[Int](new Exception("error")).getOrElse(0) == 0)
+        }
+    }
+
+    "orElse" - {
```

Comment:
Can you add a type inference test:

https://github.com/zio/zio/issues/8890

ZIO.succeed(Option.empty[List[String]]).someOrElse(List.empty)
```
Result(Option.empty[List[String]).orElse(Result(List.empty))
```

Reply:
It doesn't compile. What return type should it have? `Result[IterableOnce]`?

Reply:
Ah no, sorry that is a bad example. I forgot the `someOrElse` operator will squash the Option.

Perhaps just `Result(List.empty[String]).orElse(List.empty)`

Reply:
I've added a couple of inference tests. It seems to be working correctly. I think inference here is simpler than in ZIO's case.